ark/attest/cache/snapshots.ts

```diff
@@ -151,12 +151,18 @@ const snapshotArgsToQueuedUpdate = ({
 	baselinePath
 }: SnapshotArgs): QueuedUpdate => {
 	const snapCall = findCallExpressionAncestor(position, snapFunctionName)
-	const newArgText =
+	let newArgText =
 		typeof serializedValue === "string" && serializedValue.includes("\n") ?
 			"`" +
 			serializedValue.replaceAll("`", "\\`").replaceAll("${", "\\${") +
 			"`"
 		:	JSON.stringify(serializedValue)
+
+	if (newArgText.includes("$ark.bigint-"))
+		newArgText = `${newArgText.replace('"$ark.bigint-', "").slice(0, -1)}n`
+	else if (newArgText.includes("$ark.undefined"))
```

Comment:
Here, we should just check `===` since there is never a suffix

Reply:
So I believe we need `.includes`

This function can be called with objects that undefined is a value of:

from console logging `newArgText` like `console.log({ newArgText })`

<img width="327" alt="Screenshot 2024-09-18 at 7 25 34 PM" src="https://github.com/user-attachments/assets/87117d9c-4213-4a57-9275-292f6021db4c">

Reply:
Ahh you're right I forgot we weren't just serializing the primitive.

A couple things then:

- No need to check includes, lets just directly replace both them since that essentially checks includes implicitly
- It seems like the `bigint` case won't work as you're assuming there's no object (let's add tests for bigint and undefined being serialized in object).
- `replace` only replaces the first occurrence of a string. Let's use `replaceAll` instead with a global regex. For the bigint case you'll have to match the replacements from the matches + n, and we'll want to add tests cases for multiple undefineds/bigints being replaced.

Reply:
Added the tests. Switched to using replace with regex and the 'g' flag. I can use replaceAll instead of the g flag if you'd like though.

Reply:
I think if you use `replaceAll` it's an error if you don't also use the `/g` flag? I always use `replaceAll` but I honestly forget how it matters at this point.